File: DoctorVirtus.Aplicativo/Utils/Parse.cs
using System;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

namespace DoctorVirtus.Aplicativo.Utils {
  
    public static class Parse {

        public static object IsEmpty(string value) {
            try {
                if (value.Trim() == "")
                    return DBNull.Value;
                else
                    return value.Trim();
            } catch {
                return DBNull.Value;
            }
        }

        public static object IsEmpty(object value) {
            try {
                if (Convert.ToString(value).Trim() == "")
                    return DBNull.Value;
                else
                    return value;
            } catch {
                return DBNull.Value;
            }
        }

        /* TODO ERROR: Skipped WarningDirectiveTrivia */
        public static string ToString(object value) {
            /* TODO ERROR: Skipped WarningDirectiveTrivia */
            try {
                if (Convert.IsDBNull(value))
                    return null;
                else
                    return Convert.ToString(value).Trim();
            } catch {
                return null;
            }
        }

        public static int? ToInt(object value) {

            try {
                if (!IsNumeric(value))
                    return null;
                else
                    return Convert.ToInt32(value);
            } catch {
                return 0;
            }
        }

        public static bool ToBoolean(object value) {
            try {
                if (Convert.IsDBNull(value))
                    return false;
                else
                    return Convert.ToBoolean(value);
            } catch {
                return false;
            }
        }

        public static decimal ToDecimal(object value, int CasasDecimais = 2) {
            try {
                if (!IsNumeric(value))
                    return 0;
                else
                    return Math.Round(Convert.ToDecimal(value), CasasDecimais);// String.Format("{0:N" & CasasDecimais & "}", Convert.ToDecimal(Value))
            } catch {
                return 0;
            }
        }

        public static byte[] ToByte(object value) {
            try {
                if (Convert.IsDBNull(value))
                    return null;
                else
                    return ObjectToByteArray(value); // Value;
            } catch {
                return null;
            }
        }

        public static byte[] ObjectToByteArray(object obj) {

            if (obj == null) return null;

            BinaryFormatter bf = new BinaryFormatter();
            using (MemoryStream ms = new MemoryStream()) {
                bf.Serialize(ms, obj);
                return ms.ToArray();
            }
        }

        public static bool IsNumeric(object Expression) {
            
            if (Expression == null || Expression is DateTime) return false;
            
            if (Expression is short || Expression is int || Expression is long || Expression is decimal || Expression is float || Expression is double) return true;
            
            try {

                if (Expression is string)
                    double.Parse(Expression as string);
                else
                    double.Parse(Expression.ToString());
                return true;
            } catch {
                return false;
            }
        }
    }
}

File: DoctorVirtus.Desktop/App/Utils/Class/Parse.vb
Imports System.IO

Namespace AtlantaSistemas.Utils2.Class

    Public Class Parse

        Public Function IsEmpty(Value As String) As Object

            Try

                If Trim(Value) = "" Then
                    Return DBNull.Value
                Else
                    Return Trim(Value)
                End If

            Catch ex As Exception
                Return DBNull.Value
            End Try

        End Function

        Public Function IsEmpty(Value As Object) As Object

            Try

                If Trim(Convert.ToString(Value)) Is "" Then
                    Return DBNull.Value
                Else
                    Return Value
                End If

            Catch ex As Exception
                Return DBNull.Value
            End Try

        End Function

#Disable Warning BC40005 ' O membro sombreia um método substituível no tipo base
        Public Function ToString(Value As Object) As String
#Enable Warning BC40005 ' O membro sombreia um método substituível no tipo base

            Try

                If IsDBNull(Value) Then
                    Return Nothing
                Else
                    Return Trim(Convert.ToString(Value))
                End If

            Catch ex As Exception
                Return Nothing
            End Try

        End Function

        Public Function ToInt(Value As Object) As Integer?
            Try

                If Not IsNumeric(Value) Then
                    Return Nothing
                Else
                    Return Convert.ToInt32(Value)
                End If

            Catch ex As Exception
                Return Nothing
            End Try

        End Function

        Public Function ToBoolean(Value As Object) As Boolean

            Try

                If IsDBNull(Value) Then
                    Return False
                Else
                    Return Convert.ToBoolean(Value)
                End If

            Catch ex As Exception
                Return False
            End Try

        End Function

        Public Function ToDecimal(Value As Object, Optional CasasDecimais As Integer = 2) As Decimal

            Try

                If IsDBNull(Value) Then
                    Return 0
                Else
                    If Not IsNumeric(Value) Then
                        Return 0
                    End If

                    If IsNothing(CasasDecimais) Then
                        Return Convert.ToDecimal(Value)
                    End If

                    Return Math.Round(Value, CasasDecimais) 'String.Format("{0:N" & CasasDecimais & "}", Convert.ToDecimal(Value))

                End If

            Catch ex As Exception
                Return 0
            End Try

        End Function

        Public Function ToDateTime(Value As Object) As DateTime

            Try

                If IsDBNull(Value) Then
                    Return Nothing
                Else
                    Return Convert.ToDateTime(Value)
                End If

            Catch ex As Exception
                Return Nothing
            End Try

        End Function

        Public Function ToByte(Value As Object) As Byte()

            Try

                If IsDBNull(Value) Then
                    Return Nothing
                Else

                    Dim Arquivo As Byte() = Value

                    Return Arquivo
                End If

            Catch ex As Exception
                Return Nothing
            End Try

        End Function

    End Class

End Namespace